### PR TITLE
Load FlutterLoader when creating FlutterEngineGroup

### PR DESCRIPTION
**Pull request by xster**
Fixes https://github.com/flutter/flutter/issues/74767. Make the FlutterEngineGroup initialize the FlutterLoader. This is the same pattern as the FlutterEngine itself.  

It's a breaking change though since I need to let the FlutterEngineGroup have a context too.

### CODE REVIEW

File: shell/platform/android/test/io/flutter/embedding/engine/FlutterEngineGroupComponentTest.java [hunk 1]
```diff
@@ -127,4 +139,21 @@ public void canSpawnMoreEngines() {
...
     assertEquals(2, engineGroupUnderTest.activeEngines.size());
   }
+
+  @Test
+  public void canCreateAndRunCustomEntrypoints() {
+    FlutterEngine firstEngine =
+        engineGroupUnderTest.createAndRunEngine(
+            RuntimeEnvironment.application,
+            new DartEntrypoint(
+                FlutterInjector.instance().flutterLoader().findAppBundlePath(),
```
Comment (gaaclarke) [inline shell/platform/android/test/io/flutter/embedding/engine/FlutterEngineGroupComponentTest.java:149]: This is unfortunate, this is impossible to find for people that want to make a DartEntrypoint.  Maybe we could add a const for it in the DartEntrypoint class?   `DartEntrypoint.DEFAULT_APP_BUNDLE_PATH`?  I couldn't find an example of anyone using the new API on github.  It looks like a series of 2 changes have been deprecated and replaced for this.

**Review by gaaclarke — APPROVED** [review]
Change looks good, I think we can do better with the DartEntrypoint API though.  It's much easier on iOS because it accepts null to mean the default location.  I know we don't want to use null so maybe we can make a const?

File: shell/platform/android/test/io/flutter/embedding/engine/FlutterEngineGroupComponentTest.java (hunk 1 above)
Comment (xster) [reply shell/platform/android/test/io/flutter/embedding/engine/FlutterEngineGroupComponentTest.java:149]: Totally agree. This API makes no sense lol. But that's a "new feature" and I want to cherry pick this. Left https://github.com/flutter/flutter/issues/74828. Surprised no one complained yet :)

**Comment by xster** [comment]
Can't seem to retrigger the luci fuchsia test on head. Going to poke the infra with this PR.

**Pull request merged**